src/index.ts

```diff
@@ -19,39 +26,88 @@ interface Trie {
   put(key: Buffer | string, value: Buffer | string, cb: TriePutCb): void
 }
 
+interface AccountProps {
+  nonce: Buffer
+  balance: Buffer
+  stateRoot: Buffer
+  codeHash: Buffer
+}
+
+// Would prefer for Account class getters to have type `Buffer`
+// and setters to have type `any` to be cast `toBuffer`,
+// however get ts error:
+// `'get' and 'set' accessor must have the same type`
+type BufferLike = Buffer | any
+
 export default class Account {
+  private props: AccountProps
```

Comment:
I don't get this. Why not just private fields like `_nonce`, `_balance`, etc? Not that I'm against it either, just curious.

Reply:
I liked it more as a pattern, I'm open to either way. I only did it as a workaround for needing to define getters and setters on the properties. I think if we make the object immutable in the next breaking change revision then we can just define them as normal properties (without  needing `_` or `props`).

Reply:
Yeah, that would be great. Just set normal properties in the constructor :)